mlrun/common/schemas/partition_interval.py

```diff
@@ -114,39 +104,78 @@ def get_next_partition_time(self, current_datetime: datetime) -> datetime:
             year, week, _ = current_datetime.isocalendar()
-            return f"{year}{week:02d}"
+            return int(f"{year}{week:02d}")
         else:
             raise ValueError(f"Unsupported PartitionInterval: {self}")
 
-    def get_partition_expression(self, column_name: str):
-        if self == PartitionInterval.YEARWEEK:
-            return f"YEARWEEK({column_name}, 1)"
-        elif self == PartitionInterval.DAY:
-            # generates value in format %Y%m%d in mysql
-            # mysql query example: `select YEAR(NOW())*10000 + MONTH(NOW())*100 + DAY(NOW());`
-            return f"YEAR({column_name}) * 10000 + MONTH({column_name}) * 100 + DAY({column_name})"
-        elif self == PartitionInterval.MONTH:
-            # generates value in format %Y%m in mysql
-            # mysql query example: `select YEAR(NOW())*100 + MONTH(NOW());`
-            return f"YEAR({column_name}) * 100 + MONTH({column_name})"
-        else:
+    def get_partition_name(
+        self,
+        current_datetime: datetime,
+    ) -> str:
+        return f"p{self.get_partition_key_value(current_datetime)}"
+
+    def get_partition_expression(
+        self,
+        column_name: str,
+        dialect: str = mlrun.common.db.dialects.Dialects.MYSQL,
+    ) -> str:
+        """
+        Convert *column_name* to an integer key that works for RANGE partitioning.
+
+        Only used for legacy MySQL dialects migrations.
+        """
+
+        if dialect.startswith(mlrun.common.db.dialects.Dialects.MYSQL):
+            if self == PartitionInterval.YEARWEEK:
+                return f"YEARWEEK({column_name}, 1)"
+
+            if self == PartitionInterval.DAY:
+                return (
+                    f"YEAR({column_name}) * 10000 + "
+                    f"MONTH({column_name}) * 100 + "
+                    f"DAY({column_name})"
+                )
+
+            if self == PartitionInterval.MONTH:
+                return f"YEAR({column_name}) * 100 + MONTH({column_name})"
+
             raise ValueError(f"Unsupported PartitionInterval: {self}")
+        else:
+            raise ValueError(
+                f"Unsupported dialect: {dialect}. Supported dialects are: "
+                f"{mlrun.common.db.dialects.Dialects.MYSQL}"
+            )
 
     def get_number_of_partitions(self, days: int) -> int:
         # Calculate the number partitions based on given number of days
         if self == PartitionInterval.DAY:
             return days
         elif self == PartitionInterval.MONTH:
             # Average number days in a month is 30.44
-            return int(days / 30.44)
+            return math.ceil(days / 30.44)
         elif self == PartitionInterval.YEARWEEK:
-            return int(days / 7)
+            return math.ceil(days / 7)
         else:
             raise ValueError(f"Unsupported PartitionInterval: {self}")
+
+    @classmethod
+    def get_partition_interval_from_env(cls) -> "PartitionInterval":
```

Comment:
What if we want different partition intervals for different tables? Shouldn't we support possible env values per table name?